src/components/Transcript/TranscriptMenu/TranscriptSelector.js

```diff
@@ -1,36 +1,54 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import TranscriptDownloader from './TranscriptDownloader';
 
-const TanscriptSelector = (props) => {
-  const [title, setTitle] = React.useState(props.title);
+const MACHINE_GEN_MESSAGE = 'Machine-generated transcript may contain errors.';
+
+const TanscriptSelector = ({
+  setTranscript,
+  transcriptData,
+  transcriptInfo,
+  noTranscript,
+}) => {
+  const { title, id, tUrl, tFileExt, isMachineGen } = transcriptInfo;
+  const [currentId, setCurrentId] = React.useState(id);
 
   const selectItem = (event) => {
-    setTitle(event.target.value);
-    props.setTranscript(event.target.value);
+    setCurrentId(event.target.value);
+    setTranscript(event.target.value);
```

Comment:
Good catch! Actually the first one `setCurrentId()` can be removed because we already get relevant updated data from the parent component as `id` in the `transcriptInfo` JSON object.